Initial commit to public repo (#2)
vegacloud/vega-metrics-agent: README.md

## Key Features

- **Node Metrics**: Detailed metrics for each node, including capacity, allocatable resources, and usage.
- **Pod Metrics**: Metrics for all pods, covering resource requests, limits, and usage.
- **Cluster Metrics**: Aggregated cluster-wide metrics like total capacity, allocatable resources, and usage.
- **Persistent Volume Metrics**: Metrics for persistent volumes and persistent volume claims.
- **Namespace Metrics**: Tracks namespace resource quotas, limit ranges, and usage.
- **Workload Metrics**: Metrics for deployments, stateful sets, daemon sets, and jobs.
- **Networking Metrics**: Gathers metrics related to services and ingresses.
- **Horizontal Pod Autoscaler (HPA)**: Monitors metrics for HPAs.
- **Replication Controller & ReplicaSet Metrics**: Tracks metrics for replication controllers and replica sets.
- **Health Check**: Provides a simple health check endpoint.

## Installation

### Prerequisites

Before installing, ensure the following:

- A Kubernetes cluster
- Go 1.16+ (if building from source)
- AWS credentials (for S3 uploads)

### Building from Source

1. Clone the repository:
    ```sh

>     git clone https://github.com/vegacloud/kubernetes-metrics-agent.git
>     cd kubernetes-metrics-agent

Reviewer: ```suggestion
    git clone https://github.com/vegacloud/vega-metrics-agent.git
    cd vega-metrics-agent
```

    ```

2. Build the project:
    ```sh
    make build
    ```

3. Run the agent:
    ```sh

>     ./metrics-agent --help

Reviewer: ```suggestion
    ```sh
    ./vega-metrics-agent --help
```

- START_COLLECTION_NOW: Start metric collection immediately.
- SAVE_LOCAL: Save metrics locally.
- MAX_CONCURRENCY: Set the maximum concurrency for metric collection.

### Deploying in Kubernetes

The default namespace for the agent is \`vegacloud\`. When deployed, the agent gathers metrics from the Kubernetes API and directly from the cluster nodes. Metrics are uploaded every 10 minutes to an Amazon S3 bucket. 

Ensure that the agent has outbound access to:

- **api.vegacloud.io** (port 443) — for pre-signed URL retrieval.
- **vegametricsocean.s3.us-west-2.amazonaws.com** (port 443) — for uploading data.

If your cluster is behind a firewall, add these addresses to your outbound allowlist.

### Supported Kubernetes Versions

The Vega Kubernetes Metrics Agent supports Kubernetes versions up to 1.30 across cloud platforms such as AWS (EKS), Google Cloud (GKE), Azure (AKS), and Oracle Cloud (OKE).

### Deploying in Kubernetes using Helm Charts ###
- Install the kubectl command line tool
- Install the helm command line tool
- Modify the values.yaml file in ./deploy/charts/vega-metrics-agent/values.yaml

> - In the charts/vega-metrics-agent directory, run `helm install vega-metrics ./delpoy/charts/vega-metrics-agent` (this sets up the service account, role, and rolebinding for the agent in your cluster)

Reviewer: ```suggestion
- In the root directory, run `helm install vega-metrics ./deploy/charts/vega-metrics-agent` (this sets up the service account, role, and rolebinding for the agent in your cluster)
```
